### PR TITLE
MCP.1 fixes

### PR DESCRIPTION
**Pull request by TaranFurnell**
- Update reference model for aggregate interference to use CbsdGrantInfo
- Miscellaneous fixes for MCP.1 (from PR #617).
- Added land category calculation for GWPZ in MCP/XPR
- Corrected GWPZ file usage

### CODE REVIEW

**Comment by jni2000** [comment]
Just checked and found that:
1. "src/harness/reference_models/common/data.py" fixed the issue we encountered in #617, we took the changes on this file out of #671.
2. Changes in "src/harness/testcases/WINNF_FT_S_MCP_testcase.py" covered most of our concerns in #617 as well. Just one concern not covered. At line 626 when loading the ppa_record_1 and 2, the ppa record in testdata folder does not have ‘cbsdReferenceId’ field. For testing IAP and Aggregate models, we need this filed. We need to add this to the original test data ppa_record. We will take our changes on this "src/harness/testcases/WINNF_FT_S_MCP_testcase.py"  out and wait for your merge.
3. Will our changes in "src/harness/reference_models/pre_iap_filtering/pre_iap_util.py" in PR #617 still needed ?

Thanks.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py [hunk 1]
```diff
@@ -116,54 +118,55 @@ def executeSingleMCPIteration(self, iteration_content):
...
+    if 'gwpzRecords' in iteration_content['protectedEntities']:
+      for gwpz_record in iteration_content['protectedEntities']['gwpzRecords']:
         try:
-          self._sas_admin.InjectWisp({'record': gwpz_record})
+          self._sas_admin.InjectWisp(gwpz_record)
         except Exception as e:
           logging.error(common_strings.CONFIG_ERROR_SUSPECTED)
           raise e
-        # TODO: calculate and store the land category of the GWPZ
+        grid_points = geoutils.GridPolygon(gwpz_record['zone']['features']['geometry'], res_arcsec=2)
```
Comment (sbdt-google) [inline src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: Missing the index 0 :
```
gwpz_record['zone']['features'][0]['geometry']
```

File: src/harness/reference_models/interference/aggregate_interference.py [hunk 2]
```diff
@@ -253,13 +246,11 @@ def calculateAggregateInterferenceForPpa(ppa_record, pal_list, cbsd_list):
   Args:
     ppa_record: PPA protection entity
     pal_list: list of PAL records
-    cbsd_list: list of CBSD objects containing registrations and grants
+    cbsd_list: list of CbsdGrantInfo objects containing registrations and grants
```
Comment (sbdt-google) [inline src/harness/reference_models/interference/aggregate_interference.py]: Elsewhere in come cleanup code, all those are called simply 'grants' or 'cbsd_grants'.
But do not worry on this in this PR, I will make a cleanup PR to finalize those cosmetic things a bit later to try having most of the code consistent.

File: src/harness/reference_models/interference/aggregate_interference.py [hunk 3]
```diff
@@ -207,12 +201,11 @@ def calculateAggregateInterferenceForGwpz(gwpz_record, cbsd_list):
...
   Args:
     gwpz_record: GWPZ protection entity
-    cbsd_list: list of CBSD objects containing registrations and grants
+    cbsd_list: list of CbsdGrantInfo objects containing registrations and grants
   Returns:
     Aggregate interference to GWPZ in the nested dictionary format.
       {latitude : {longitude: [aggr_interf1(mW), ..., aggr_interfK(mW)]}}
   """
-  grant_list = data.getAllGrantInfoFromCbsdDataDump(cbsd_list)
   gwpz_region = gwpz_record['zone']['features'][0]['properties']['clutter']
```
Comment (jni2000) [inline src/harness/reference_models/interference/aggregate_interference.py]: 'clutter' shall be changed to 'landCategory' as per discussion with Kate and Serge.

File: src/harness/testcases/WINNF_FT_S_MCP_testcase.py (hunk 1 above)
Comment (TaranFurnell) [reply src/harness/testcases/WINNF_FT_S_MCP_testcase.py]: Done.

File: src/harness/reference_models/interference/aggregate_interference.py (hunk 3 above)
Comment (TaranFurnell) [reply src/harness/reference_models/interference/aggregate_interference.py]: Done.

**Comment by TaranFurnell** [comment]
The preIAP fix looks correct, I've added it to this PR. I don't know what the correct data format for the other changes are, will leave that to Kate.

**Pull request merged**